More dynamicDowncast and RefPtr in WebCore/style

annevk opened the pull request:
#### bfcf7fa3af7e2a898dffe289be9549f3dc7853eb
<pre>
More dynamicDowncast and RefPtr in WebCore/style
<a href="https://bugs.webkit.org/show_bug.cgi?id=266208">https://bugs.webkit.org/show_bug.cgi?id=266208</a>

Reviewed by Chris Dumez.

This leaves one downcast in StyleAdjuster alone that is guarded by
ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION).

* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::shouldInheritTextDecorationsInEffect):
(WebCore::Style::Adjuster::adjust const):
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
(WebCore::Style::isOutermostSVGElement): Deleted.
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveElement):
(WebCore::Style::TreeResolver::pushParent):
(WebCore::Style::TreeResolver::resolveComposedTree):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):

Canonical link: <a href="https://commits.webkit.org/271885@main">https://commits.webkit.org/271885@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5f61b0b6e9f3298bee09f1b303931517842e3c8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29933 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8599 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31251 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32445 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27068 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30593 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10785 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5854 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27118 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30227 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7206 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25538 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6148 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6317 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26627 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33781 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27302 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27052 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32491 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6250 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4429 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30281 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7986 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7094 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6992 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6776 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/style/StyleAdjuster.cpp:
```diff
@@ -478,7 +479,7 @@ void Adjuster::adjust(RenderStyle& style, const RenderStyle* userAgentAppearance
         // must be positioned for z-index to apply to them.
         if (element && element->document().settings().layerBasedSVGEngineEnabled()) {
-            if (auto* svgElement = dynamicDowncast<SVGElement>(element); svgElement && svgElement->isOutermostSVGSVGElement())
+            if (isOutermostSVGElement(element))
```
Not new but we're doing an unnecessary null check of `element` here.

annevk replied:
Fixed, but I was not able to reuse the static so there's some minor duplication (that is also not new, to be clear).

annevk committed:
More dynamicDowncast and RefPtr in WebCore/style
https://bugs.webkit.org/show_bug.cgi?id=266208

Reviewed by Chris Dumez.

This leaves one downcast in StyleAdjuster alone that is guarded by
ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION).

* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::shouldInheritTextDecorationsInEffect):
(WebCore::Style::Adjuster::adjust const):
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
(WebCore::Style::isOutermostSVGElement): Deleted.
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveElement):
(WebCore::Style::TreeResolver::pushParent):
(WebCore::Style::TreeResolver::resolveComposedTree):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):

Canonical link: https://commits.webkit.org/271885@main
